packages/api/orm/migration/1638312144983-initial.ts

```diff
@@ -0,0 +1,232 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class initial1638312144983 implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    const sql = `    
+    SET statement_timeout = 0;
+    SET lock_timeout = 0;
+    SET idle_in_transaction_session_timeout = 0;
+    SET client_encoding = 'UTF8';
+    SET standard_conforming_strings = on;
+    SELECT pg_catalog.set_config('search_path', '', false);
+    SET check_function_bodies = false;
+    SET xmloption = content;
+    SET client_min_messages = warning;
+    SET row_security = off;
+    
+    CREATE EXTENSION IF NOT EXISTS "uuid-ossp" WITH SCHEMA public;
+    
+    COMMENT ON EXTENSION "uuid-ossp" IS 'generate universally unique identifiers (UUIDs)';
+    
+    CREATE TYPE public.application_provisioningstatus_enum AS ENUM (
+        'not_started',
+        'in_progress',
+        'failed',
+        'complete'
+    );
+    
+    
+    ALTER TYPE public.application_provisioningstatus_enum OWNER TO postgres;
+    
+    CREATE TYPE public.application_status_enum AS ENUM (
+        'not_started',
+        'in_progress',
+        'failed',
+        'complete'
+    );
+    
+    ALTER TYPE public.application_status_enum OWNER TO postgres;
+    
+    CREATE TYPE public.environment_provisioningstatus_enum AS ENUM (
+        'not_started',
+        'in_progress',
+        'failed',
+        'complete'
+    );
+    
+    ALTER TYPE public.environment_provisioningstatus_enum OWNER TO postgres;
+    
+    CREATE TYPE public.environment_status_enum AS ENUM (
+        'not_started',
+        'in_progress',
+        'failed',
+        'complete'
+    );
+    
+    ALTER TYPE public.environment_status_enum OWNER TO postgres;
+    
+    CREATE TYPE public.portfolio_csp_enum AS ENUM (
+        'CSP A',
+        'CSP B'
+    );
+    
+    ALTER TYPE public.portfolio_csp_enum OWNER TO postgres;
```

Comment:
How miserable is it to migrate enum's later? Let's say CSPs are later "Foo" and "Bar". We're looking at a migration that:

1. Adds the new variants
2. Does something with all As and Bs (either renaming to Foo or Bar or dropping them)
3. Deletes the old variants

?

Reply:
Do you suggest finding a different approach for the enums due to complex migration changes, or that we should think a bit more before we go down this path? Doing a quick search I found [PostgreSQL: ENUM is no Silver Bullet](https://medium.com/swlh/postgresql-3-ways-to-replace-enum-305861e089bc) that give alternate options that we can consider.

Reply:
Interesting food for thought, but I am still inclined to stick with `TYPE` rather than any of those options or build the validation at the app layer. If the values do need to change then we'll have to fix it with a migration regardless, but with a `TYPE` we can be sure that before our code even executes that our existing data set will be valid.

Reply:
I think we've decided that for now we're set on `enum`s. But I just want to make sure we have some idea of what we're going to do for this since we _know_ that that we're eventually going to have to.